google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java

```diff
@@ -0,0 +1,367 @@
+            // child accounts for printing the hierarchy afterwards.
+            customerIdsToChildAccounts.put(customerId, customerClient);
+            // Checks if the child account is a manager itself so that it can later be processed
+            // and added to the map if it hasn't been already.
+            if (customerClient.getManager().getValue()) {
+              // A customer can be managed by multiple managers, so to prevent visiting the same
+              // customer multiple times, we need to check if it's already in the map.
+              boolean alreadyVisited =
+                  customerIdsToChildAccounts.containsKey(customerClient.getId().getValue());
+              if (!alreadyVisited && customerClient.getLevel().getValue() == 1) {
+                managerAccountsToSearch.add(customerClient.getId().getValue());
+              }
+            }
+          }
+        } catch (GoogleAdsException gae) {
+          System.out.printf(
+              "Unable to retrieve hierarchy for customer ID %d: %s%n",
+              customerId, gae.getGoogleAdsFailure().getErrors(0).getMessage());
+        }
+      }
+
+      // The rootCustomerClient will be null if the account hierarchy was unable to be retrieved
+      // (e.g. the account is a test account or a client account with an incomplete billing setup.
+      // This method returns null in these cases to add the seedCustomerId to the list of
+      // customer IDs for which the account hierarchy could not be retrieved.
+      if (rootCustomerClient == null) {
+        return null;
+      }
+
+      Map<CustomerClient, Multimap<Long, CustomerClient>> CustomerClientToHierarchy =
+          new HashMap<>();
+      CustomerClientToHierarchy.put(rootCustomerClient, customerIdsToChildAccounts);
+      return CustomerClientToHierarchy;
+    }
+  }
+
+  /**
+   * Retrieves a list of accessible customers with the provided set up credentials.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @return a list of customer IDs.
+   */
+  private List<Long> getAccessibleCustomers(GoogleAdsClient googleAdsClient) {
+    List<Long> seedCustomerIds = new ArrayList<>();
+    // Issues a request for listing all accessible customers by this authenticated Google account.
+    try (CustomerServiceClient customerServiceClient =
+        googleAdsClient.getLatestVersion().createCustomerServiceClient()) {
+      ListAccessibleCustomersResponse accessibleCustomers =
+          customerServiceClient.listAccessibleCustomers(
+              ListAccessibleCustomersRequest.newBuilder().build());
+      System.out.println(
+          "No manager customer ID is specified. The example will print the "
+              + "hierarchies of all accessible customer IDs:");
+
+      for (String customerResourceName : accessibleCustomers.getResourceNamesList()) {
+        long customer = Long.parseLong(CustomerName.parse(customerResourceName).getCustomer());
```

Comment:
Like the above comment, we should probably declare this as Long, not long, (and use Long.valueOf() instead) to avoid unnecessary unboxing processes.
The GoogleAdsClient mostly also uses the boxed version, so I don't see a need to use unboxed version here too.

Reply:
We typically use the primitive version unless the boxed version is necessary elsewhere in the library.